### PR TITLE
Adds install info with Chocolatey

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Creates `install_info` file when installing using Chocolatey, adding information about the Chocolatey version, Agent version and whether it was an online or offline install.

### Motivation

- Allow users to track their installation methods
- Allows Datadog to make stats on installation methods

### Additional Notes

The installer version includes whether the online/offline version of the Chocolatey installer was used. If this setup is temporary we should think if this information is really needed and what is the best way to present it.

### Describe your test plan

Tested that the file was created during installation and contained the relevant information using the [Chocolatey test environment](https://github.com/chocolatey-community/chocolatey-test-environment).

### CODE REVIEW

File: chocolatey/tools-offline/chocolateyinstall.ps1 [hunk 1]
```diff
@@ -21,3 +21,11 @@ $packageArgs = @{
   validExitCodes= @(0, 3010, 1641)
 }
 Install-ChocolateyInstallPackage @packageArgs
+
+$installInfo = @"
+tool: chocolatey
+tool_version: chocolatey-$($env:CHOCOLATEY_VERSION)
+installer_version: $($env:chocolateyPackageVersion)-offline
```
Comment (albertvaka) [inline chocolatey/tools-offline/chocolateyinstall.ps1:28]: What's the format of `$env:chocolateyPackageVersion`? If it's just a version number, we might want to prefix it with `chocolatey-` to follow the same format as in other places.

File: chocolatey/tools-online/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -16,3 +16,11 @@ $packageArgs = @{
...
 }
 Install-ChocolateyPackage @packageArgs
+
+$installInfo = @"
+tool: chocolatey
+tool_version: chocolatey-$($env:CHOCOLATEY_VERSION)
+installer_version: $($env:chocolateyPackageVersion)-online
+"@
+
+Out-File -FilePath C:\ProgramData\Datadog\install_info -InputObject $installInfo
```
Comment (albertvaka) [inline chocolatey/tools-online/chocolateyinstall.ps1:26]: Under which user is this file created? We need it to be readable (owned, probably) by `ddagentuser`.

File: chocolatey/tools-offline/chocolateyinstall.ps1 [hunk 3]
```diff
@@ -21,3 +21,11 @@ $packageArgs = @{
   validExitCodes= @(0, 3010, 1641)
 }
 Install-ChocolateyInstallPackage @packageArgs
+
+$installInfo = @"
+tool: chocolatey
```
Comment (albertvaka) [inline chocolatey/tools-offline/chocolateyinstall.ps1:26]: I think we expect these three fields to not be at the top level, but within a `install_method` field. 

Using this as a reference: https://github.com/DataDog/datadog-agent/pull/5414/files#diff-ee095aa3ebc9c402027d28eef39fd88bR349

@kbogtob to confirm.

Comment (mx-psi) [reply chocolatey/tools-offline/chocolateyinstall.ps1:26]: I think you are right (waiting on @kbogtob to confirm), I misread the Jira card.

File: chocolatey/tools-online/chocolateyinstall.ps1 (hunk 2 above)
Comment (mx-psi) [reply chocolatey/tools-online/chocolateyinstall.ps1:26]: `ddagentuser` has `FullControl` over the `install_info` file but it is not the owner, but `ddagentuser` is not the owner of `datadog.yaml` either. 

For reference here are the owners and access of each of those files on a test run on my testing environment:

```
Path         Owner                  Access                                     
----         -----                  ------                                     
datadog.yaml NT AUTHORITY\SYSTEM    NT AUTHORITY\SYSTEM Allow  FullControl     
                                    BUILTIN\Administrators Allow  FullControl  
                                    BUILTIN\Users Allow  ChangePermissions     
                                    WIN2012R2X64\ddagentuser Allow  FullControl
install_info BUILTIN\Administrators WIN2012R2X64\ddagentuser Allow  FullControl
                                    BUILTIN\Users Allow  ChangePermissions     
                                    BUILTIN\Administrators Allow  FullControl  
                                    NT AUTHORITY\SYSTEM Allow  FullControl     
```

Comment (albertvaka) [reply chocolatey/tools-online/chocolateyinstall.ps1:27]: 👍 Makes sense, since it inherits it from the parent directory.

**Review by albertvaka — APPROVED** [review]

**Pull request merged**